specification/cognitiveservices/OpenAI.Inference/models/completions/chat_completions.tsp

```diff
@@ -121,10 +118,22 @@ model ChatCompletionsOptions {
   @added(ServiceApiVersions.v2023_07_01_Preview)
   functionCall?: FunctionCallConfig;
 
-  @doc("The maximum number of tokens to generate.")
+  /**
+   * The maximum number of tokens allowed for the generated answer. By default, the number of tokens the model can return will be (4096 - prompt tokens).
+   *
+   * This value is now deprecated in favor of `max_completion_tokens`, and is not compatible with o1 series models.
+   */
+  #deprecated "Field deprecated since v2024-09-01-preview service version."
   @encodedName("application/json", "max_tokens")
   maxTokens?: int32;
 
+  /**
+   * An upper bound for the number of tokens that can be generated for a completion, including visible output tokens and reasoning tokens.
+   */
+  @encodedName("application/json", "max_completion_tokens")
+  @added(ServiceApiVersions.v2024_09_01_Preview)
+  maxCompletionTokens?: int32;
```

Comment:
Just as a note, I'm following up right now for information on why `max_completion_tokens` appears to still be rejected with `2024-09-01-preview` despite being in the spec. I'm concerned that pre-o1 models may not be configured to deal with this yet, which would be a critical parity gap and *really* hard problem for SDK libraries to solve.

At the time of writing, this failed with an "unrecognized parameter" error:
```
curl https://openai-sdk-test-automation-eastus2.openai.azure.com/openai/deployments/gpt-4o/chat/completions?api-version=2024-09-01-preview \
  -H "api-key: $AOAI_EUS2_API_KEY" \
  -H "Content-Type: application/json" \
  -d '{
    "messages": [{"role":"user","content":"hello, assistant!"}],
    "max_completion_tokens": 1024
  }'
```